ggml/src/ggml-cpu/ggml-cpu-quants.c

```diff
@@ -1759,66 +1759,76 @@ void ggml_vec_dot_q4_0_q8_0(int n, float * restrict s, size_t bs, const void * r
         const block_q8_0 * restrict vy0 = vy;
         const block_q8_0 * restrict vy1 = (const block_q8_0 *) ((const uint8_t*)vy + by);
 
-        float32x4_t sumv0 = vdupq_n_f32(0.0f);
+        if (ggml_cpu_has_matmul_int8()) {
```

Comment:
We need to remove the ARM runtime feature detection completely, it doesn't work at all and never will. So I would prefer if at least we don't make that task worse by adding more checks like this.

Reply:
Ok, I will change the PR to just include the F16 -> F32 fix in the `Q4_1` kernel.